File: CV.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace cv;
using namespace std;

int main() {
    // Load the pre-trained Haar Cascade face detector
    CascadeClassifier faceCascade;
    String faceCascadePath = "haarcascade_frontalface_default.xml";
    
    if (!faceCascade.load(faceCascadePath)) {
        cerr << "Error loading face cascade file" << endl;
        return -1;
    }

    // Initialize video capture from default camera
    VideoCapture cap(0);
    
    if (!cap.isOpened()) {
        cerr << "Error opening video stream" << endl;
        return -1;
    }

    Mat frame;
    namedWindow("Face Detection", WINDOW_AUTOSIZE);

    while (true) {
        // Capture frame-by-frame
        cap >> frame;

        // If the frame is empty, break immediately
        if (frame.empty())
            break;

        // Convert to grayscale
        Mat grayFrame;
        cvtColor(frame, grayFrame, COLOR_BGR2GRAY);

        // Detect faces
        vector<Rect> faces;
        faceCascade.detectMultiScale(grayFrame, faces, 1.1, 5, 0, Size(30, 30));

        // Draw rectangles around the faces
        for (const Rect& face : faces) {
            rectangle(frame, face, Scalar(0, 255, 0), 2);
        }

        // Display the resulting frame
        imshow("Face Detection", frame);

        // Press ESC to exit
        if (waitKey(1) == 27)
            break;
    }

    // Release the video capture object and close windows
    cap.release();
    destroyAllWindows();

    return 0;
}

File: CV.cs
using OpenCvSharp;
using System;
using System.Collections.Generic;

class FaceDetection
{
    static void Main()
    {
        // Load the pre-trained Haar Cascade face detector
        var faceCascade = new CascadeClassifier();
        if (!faceCascade.Load("haarcascade_frontalface_default.xml"))
        {
            Console.WriteLine("Error loading face cascade file");
            return;
        }

        // Initialize video capture from default camera
        using var capture = new VideoCapture(0);
        if (!capture.IsOpened())
        {
            Console.WriteLine("Error opening video stream");
            return;
        }

        using var window = new Window("Face Detection");
        using var frame = new Mat();

        while (true)
        {
            // Capture frame-by-frame
            capture.Read(frame);

            // If the frame is empty, break immediately
            if (frame.Empty())
                break;

            // Convert to grayscale
            using var grayFrame = new Mat();
            Cv2.CvtColor(frame, grayFrame, ColorConversionCodes.BGR2GRAY);

            // Detect faces
            Rect[] faces = faceCascade.DetectMultiScale(
                grayFrame,
                scaleFactor: 1.1,
                minNeighbors: 5,
                flags: HaarDetectionTypes.ScaleImage,
                minSize: new Size(30, 30)
            );

            // Draw rectangles around the faces
            foreach (var face in faces)
            {
                Cv2.Rectangle(frame, face, Scalar.Green, thickness: 2);
            }

            // Display the resulting frame
            window.ShowImage(frame);

            // Press ESC to exit
            if (Cv2.WaitKey(1) == 27)
                break;
        }
    }
}